src/mlio/default_instance_reader.cxx

```diff
@@ -158,20 +161,58 @@ default_instance_reader::handle_nested_errors()
 std::optional<memory_slice>
 default_instance_reader::read_record_payload()
 {
+    std::string split_record_err_str = "Split record is not formatted with the correct cflag.";
+    if (this->corrupt_split_record_) {
+        throw corrupt_record_error(split_record_err_str); 
+    }
     std::optional<record> rec = read_record();
     if (rec == std::nullopt) {
         return {};
     }
 
-    if (rec->kind() != record_kind::complete) {
-        throw not_supported_error{"Split records are not supported yet!"};
-    }
+    if (rec->kind() == record_kind::complete) {
+        num_bytes_read_ += rec->size();
+        store_record_idx_++;
+
+        return std::move(*rec).payload();
+    } else {
```

Comment:
No need for an `else` here as the `if` part ends with a return statement.

Reply:
Removed with the reorg from below.